### PR TITLE
TASK: Get rid of PropertyMappingConfigurationBuilder

### PR DESCRIPTION
**Pull request by kitsunet**
The ``PropertyMappingConfigurationBuilder`` its superfluous and contains
only one method that can easily be integrated into the ``PropertyMapper`` as
you will need that anyway to do something with the configuration.

Therefore ``PropertyMappingConfigurationBuilder::build()`` is deprecated
and fully replaced by ``PropertyMapper::buildPropertyMappingConfiguration()``.
The ``PropertyMappingConfigurationBuilder`` will be removed for the next
major Flow version.

### CODE REVIEW

File: TYPO3.Flow/Classes/TYPO3/Flow/Property/PropertyMapper.php [hunk 1]
```diff
@@ -104,7 +104,7 @@ public static function getTypeConverterImplementationClassNames($objectManager)
     public function convert($source, $targetType, \TYPO3\Flow\Property\PropertyMappingConfigurationInterface $configuration = null)
     {
         if ($configuration === null) {
-            $configuration = $this->configurationBuilder->build();
```
Comment (daniellienert) [inline TYPO3.Flow/Classes/TYPO3/Flow/Property/PropertyMapper.php]: The configurationBuilder injection should also be removed.

Comment (kitsunet) [inline TYPO3.Flow/Classes/TYPO3/Flow/Property/PropertyMapper.php]: Right!

Comment (kitsunet) [inline TYPO3.Flow/Classes/TYPO3/Flow/Property/PropertyMapper.php]: Done

**Comment by lorenzulrich** [comment]
It looks as if the documentation needs to be adjusted here:
http://flowframework.readthedocs.io/en/stable/TheDefinitiveGuide/PartIII/PropertyMapping.html#configuring-the-conversion-process

**Pull request merged**